src/container/srv_target.c

```diff
@@ -822,7 +822,18 @@ ds_cont_tgt_open(uuid_t pool_uuid, uuid_t cont_hdl_uuid,
 		DP_UUID(pool_uuid), DP_UUID(cont_uuid), DP_UUID(cont_hdl_uuid));
 
 	rc = dss_thread_collective(cont_open_one, &arg, 0);
-	D_ASSERTF(rc == 0, "%d\n", rc);
+	if (rc != 0) {
+		/* Once it exclude the target from the pool, since the target
+		 * might still in the cart group, so IV cont open might still
+		 * come to this target, especially if cont open/close will be
+		 * done by IV asynchronously, so this cont_open_one might return
+		 * -DER_NO_HDL sometimes.
```

Comment:
This comment should say explicitly that the -DER_NO_HDL is for the pool handle.

Reply:
then we just fail the container open? It sounds weird because we are supposed to succeed right?

Reply:
This mostly due to the inconsistent between pool map memship and pool cart group member.  See my comment here. Maybe it will be fixed once li wei fix this.  But now if the user do pool exclude, then open container, it will fail, see DAOS-3185

Reply:
Btw: why this collective call has to succeeds? why it is Assert(rc == 0) initially?  I saw there is a lot places in collective open might fail.

Reply:
I'm not saying we should access(rc == 0), what I was trying to say is, if the target is evicted from the pool, we shouldn't fail the container open just because the excluded target can't open?

Reply:
ah, you mean we should exclude the DOWN target for container open. Hmm, probably we should. the only problem is we do not promise each target has updated pool map. 
Li Wei any comments here?

Reply:
I think both "excluding DOWN targets" and "updated pool map" are good points. We need somehow excluding DOWN targets using the latest pool map.